### ISSUE TITLE
iOS build fail due to errors from win32 and ffi packages 

### ISSUE THREAD

**Issue by osaxma** [flutter/flutter#76974]
I'm unable to build any iOS app that has win32 or ffi in the dependency tree using latest flutter master `1.27.0-9.0.pre.129`.  

## Steps to Reproduce

1. Create Flutter project
2. add `path_provider: 1.6.27` in `pubspec.yaml` or any package that  has `win32` as a dependency. 
3. import the package in `main.dart`:
```dart
import 'package:path_provider/path_provider.dart';

void main() {
  runApp(MyApp()); // counterApp 
}

``` 
4. run the app. 

**Error** 

```
Launching lib/main.dart on iPhone 12 Pro Max in debug mode...
 lib/main.dart
Xcode build done.                                           11.6s
Failed to build iOS app
Error output from Xcode build:
↳
    ** BUILD FAILED **
Xcode's output:
↳
    ../../.pub-cache/hosted/pub.dartlang.org/win32-1.7.4+1/lib/src/structs.dart:1111:7: Error: Struct 'ENUMLOGFONTEX' is empty. Empty structs are undefined behavior.
    class ENUMLOGFONTEX extends Struct {
          ^
    ../../.pub-cache/hosted/pub.dartlang.org/win32-1.7.4+1/lib/src/structs.dart:2835:7: Error: Struct 'BLUETOOTH_PIN_INFO' is empty. Empty structs are undefined behavior.
    class BLUETOOTH_PIN_INFO extends Struct {
          ^
    ../../.pub-cache/hosted/pub.dartlang.org/win32-1.7.4+1/lib/src/structs.dart:2960:7: Error: Struct 'EXCEPINFO' is empty. Empty structs are undefined behavior.
    class EXCEPINFO extends Struct {}
          ^
    ../../.pub-cache/hosted/pub.dartlang.org/win32-1.7.4+1/lib/src/structs.dart:2966:7: Error: Struct 'PROPERTYKEY' is empty. Empty structs are undefined behavior.
    class PROPERTYKEY extends Struct {}
          ^
    ../../.pub-cache/hosted/pub.dartlang.org/win32-1.7.4+1/lib/src/structs.dart:2973:7: Error: Struct 'PROPVARIANT' is empty. Empty structs are undefined behavior.
    class PROPVARIANT extends Struct {}
          ^
    ../../.pub-cache/hosted/pub.dartlang.org/win32-1.7.4+1/lib/src/structs.dart:2978:7: Error: Struct 'SAFEARRAY' is empty. Empty structs are undefined behavior.
    class SAFEARRAY extends Struct {}
          ^
    ../../.pub-cache/hosted/pub.dartlang.org/win32-1.7.4+1/lib/src/structs.dart:2985:7: Error: Struct 'CLSID' is empty. Empty structs are undefined behavior.
    class CLSID extends Struct {}
          ^
    ../../.pub-cache/hosted/pub.dartlang.org/win32-1.7.4+1/lib/src/structs.dart:2992:7: Error: Struct 'STATSTG' is empty. Empty structs are undefined behavior.
    class STATSTG extends Struct {}
          ^
    ../../.pub-cache/hosted/pub.dartlang.org/win32-1.7.4+1/lib/src/structs.dart:2999:7: Error: Struct 'NLM_SIMULATED_PROFILE_INFO' is empty. Empty structs are undefined behavior.
    class NLM_SIMULATED_PROFILE_INFO extends Struct {}
          ^
    ../../.pub-cache/hosted/pub.dartlang.org/ffi-0.1.3/lib/src/utf8.dart:23:7: Error: Struct 'Utf8' is empty. Empty structs are undefined behavior.
    class Utf8 extends Struct {
          ^
    ../../.pub-cache/hosted/pub.dartlang.org/ffi-0.1.3/lib/src/utf16.dart:16:7: Error: Struct 'Utf16' is empty. Empty structs are undefined behavior.
    class Utf16 extends Struct {
          ^
    ../../.pub-cache/hosted/pub.dartlang.org/ffi-0.1.3/lib/src/allocation.dart:47:33: Error: Expected type 'T' to be a valid and instantiated subtype of 'NativeType'.
      final int totalSize = count * sizeOf<T>();
                                    ^
    Command PhaseScriptExecution failed with a nonzero exit code
    note: Using new build system
    note: Building targets in parallel
    note: Planning build
    note: Constructing build description
Could not build the application for the simulator.
Error launching application on iPhone 12 Pro Max.
Exited (sigterm)

```

<details>
  <summary>flutter doctor -v</summary>

```
[✓] Flutter (Channel master, 1.27.0-9.0.pre.129, on Mac OS X 10.15.6 19G2021 darwin-x64, locale en)
    • Flutter version 1.27.0-9.0.pre.129 at /Users/osaxma/fvm/versions/master
    • Framework revision fc77610dd6 (79 minutes ago), 2021-03-01 14:06:03 +0530
    • Engine revision 34236c9837
    • Dart version 2.13.0 (build 2.13.0-81.0.dev)

[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.2)
    • Android SDK at /Users/osaxma/Library/Android/sdk
    • Platform android-29, build-tools 29.0.2
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.3, Build version 12C33
    • CocoaPods version 1.10.1

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)

[✓] VS Code (version 1.53.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.19.0

[✓] Connected device (2 available)
    • iPhone 12 Pro Max (mobile) • 5D894941-88ED-47BA-8FE0-0FA45101E676 • ios            •
      com.apple.CoreSimulator.SimRuntime.iOS-14-3 (simulator)
    • Chrome (web)               • chrome                               • web-javascript • Google Chrome 88.0.4324.192

• No issues found!

```

</details>

**Comment by dcharkes** [comment]
Duplicate of #76705, the dependencies need to be rolled forward.

**Comment by osaxma** [comment]
@dcharkes the dependency in question, `path_provider` is a flutter plugin and I'm using the latest version `1.6.27` from pub.dev ...  How am I suppose to roll it forward?

**Comment by dcharkes** [comment]
You can use https://pub.dev/packages/path_provider/versions/2.0.0.
